src/main/kotlin/com/jetbrains/snakecharm/SnakemakeRuleInheritanceMarkerProvider.kt

```diff
@@ -31,9 +31,11 @@ class SnakemakeRuleInheritanceMarkerProvider : RelatedItemLineMarkerProvider() {
         use: SmkUse,
         result: MutableCollection<in RelatedItemLineMarkerInfo<*>>
     ) {
-        val name = use.getIdentifierLeaf() ?: return
+        val name = (if (use.nameIdentifierIsWildcard()) use.nameIdentifier?.firstChild else use.nameIdentifier) ?: return
+
         val results =
-            use.getImportedRuleNames()?.mapNotNull { it.reference.resolve() } // Inherited rules are declared in list
+            use.getDefinedReferencesOfImportedRuleNames()
+                ?.mapNotNull { it.reference.resolve() } // Inherited rules are declared in list
```

Comment:
This call chain is very hard to read and understand. Please refactor it as not a single call chain, so it should be easier to read and understand in which condition what is used.